common/src/main/java/org/keycloak/common/enums/SslRequired.java

```diff
@@ -51,10 +51,23 @@ public boolean isRequired(String address) {
     private boolean isLocal(String remoteAddress) {
         try {
             InetAddress inetAddress = InetAddress.getByName(remoteAddress);
-            return inetAddress.isAnyLocalAddress() || inetAddress.isLoopbackAddress() || inetAddress.isSiteLocalAddress();
+            return inetAddress.isAnyLocalAddress() || inetAddress.isLoopbackAddress() || inetAddress.isSiteLocalAddress() || inetAddress.isLinkLocalAddress() || isUniqueLocal(inetAddress);
```

Comment:
I think call to `InetAddress.isAnyLocalAddress()` is redundant. Maybe the thinking was that `isAnyLocalAddress()` would return `true` for any clients that communicate using IP address of any local interface - but this is not what happens. It just checks for special `0.0.0.0` IP address (see [here](https://github.com/openjdk/jdk17u/blob/c9d83d392f3809bf536afdcb52142ee6916acff0/src/java.base/share/classes/java/net/Inet4Address.java#L172-L174) and [here](https://github.com/openjdk/jdk17u/blob/c9d83d392f3809bf536afdcb52142ee6916acff0/src/java.base/share/classes/java/net/Inet6Address.java#L310-L316)). That is `INADDR_ANY` ([link](https://man7.org/linux/man-pages/man7/ip.7.html)) which is used only for socket binding. As far as I can see, it cannot ever be peer address. 

But for now, I did not remove it since it has been always here. If you agree with above, I can remove it.